src/librustdoc/html/render.rs

```diff
@@ -545,7 +546,10 @@ pub fn run(mut krate: clean::Crate,
         issue_tracker_base_url: None,
         layout: layout::Layout {
             logo: String::new(),
-            favicon: String::new(),
+            favicon: favicon.as_ref()
+                            .map(|_| format!("favicon-{}.ico", krate.name))
```

Comment:
Modifying the favicon filename/URL in the doc output will not work for docs.rs, since it will break the caching that is already done for the `/favicon.ico` URL.